### PR TITLE
Improve the construction of the praise regex + happy birthday command + Japanese praise

### PR DESCRIPTION
**Pull request by Luffyyy**
Birthday command only works between the 14th of August to 17th UTC time.
Satania will react now to satania the best / satania best / satania best girl/demon/debiru/devil/akuma/daiakuma (same as praise)

### CODE REVIEW

File: src/utils/regex-constants.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const reg = require('./reg');
+
+const kurumizawa = /(kurumizawa|胡桃沢)?\s*/;
+const satania = /(satan(i|y|ichi)a|Сатании|サタニキア|サターニャ)\s*-?/;
+const mcdowell = /(mcdowellu?|マクドウェル)?\s*/;
+const honorifics = /(sama|san|chan|senpai|さま|様|さん|ちゃん|せんぱい|先輩)?\s*/;
+const extra = /(!+|！+|¡+|﹗+|︕+|‼+|¿+|？+|⁉+|﹖+|︖+|⁈+|⁇+|\?+|~+|˜+|～+|〜+)?/;
```
Comment (Pizzacus) [inline src/utils/regex-constants.js]: This could probably be a regex group like `[!！¡﹗︕‼¿？⁉﹖︖⁈⁇?~˜～〜]`

File: src/commands/bday.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		reg`${frenchBirthday},?\s*${sataniaName}`,
+		reg`${sataniaName},?\s*${frenchBirthday}`
+	],
+	portuguese: [
+		reg`${portugueseBirthday},?\s*${sataniaName}`,
+		reg`${sataniaName},?\s*${portugueseBirthday}`
+	]
+};
+
+const allSentences = Object.values(languages).reduce((collected, current) => [...collected, ...current], []);
```
Comment (Pizzacus) [inline src/commands/bday.js]: That feels like a lot of work and complexity

The data could already be formatted correctly. You could just have all regexes in one array and indicate the languages with code comments

File: src/commands/praise.js [hunk 3]
```diff
@@ -1,10 +1,28 @@
...
 
-const praiseRegex = /^(Слава|praise|just|love?|(all )?h(a|e)il|Sauda(ç|c)(õ|o)es a|a(ç|c)ai|salve|gloire? (à|a)|(alaben|rezen|diosa) a?)?\s*(kurumizawa\s*)?(satan(i|y|ichi)a|Сатании)\s*-?(sama|san|chan|senpai)?\s*(McDowellu?\s*)?((the\s*)?best\s*)?(!+|\?+)?$/i;
+const languages = {
+	english: [
+		reg`(praise|just|love|(all )?h(a|e)il)?\s*${sataniaName}`,
+		reg`${sataniaName}\s*(praise|love|(all )?h(a|e)il)?\s*`,
+		reg`${sataniaName}\s*(the\s*)?best\s*((girl|debiru|devil|demon|(dai)?akuma)\s*)?`
+	],
+	japanese: [
+		reg`${sataniaName}${/(((のこと|の事)?(は|が)?((だい)?すき|大?好き|世界一|一番|かわいい|可愛い|最高)(だ|です)?)|を?褒めて(ください)?)?/}${/よ?ね?/}`
```
Comment (Pizzacus) [inline src/commands/praise.js]: is those two last `${}` for syntax highlighting? if so, fair enough I guess ^^

File: src/utils/reg.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+
+function isIterator(val) {
+	return val && typeof val[Symbol.iterator] === 'function';
```
Comment (Pizzacus) [inline src/utils/reg.js]: This check will fail if `val` is an empty array or string, as they evaluate to false, while they are iterators... You should use `val != null` instead

File: src/utils/regex-constants.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+const sataniaName = reg`${kurumizawa}${satania}${mcdowell}${honorifics}`;
+
+module.exports = {
```
Comment (Pizzacus) [inline src/utils/regex-constants.js:10]: you could use the shorthand notation, like `{ kurumizawa, satania, mcdowell, honorifics, extra, sataniaName }`

File: src/utils/regex-constants.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+const reg = require('./reg');
+
+const kurumizawa = /(kurumizawa|胡桃沢)?\s*/;
+const satania = /(satan(i|y|ichi)a|Сатании|サタニキア|サターニャ)\s*-?/;
```
Comment (Pizzacus) [inline src/utils/regex-constants.js]: Could you also add "사타냐" and "萨塔妮娅" (From korean and chinese satania.moe), sorry for forgetting ^^'

File: src/utils/reg.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	if (Array.isArray(strings)) {
+		return processTemplate(strings, ...keys);
+	}
+
+	return processTemplate;
+}, {
+	get: (obj, key) => {
+		if (typeof key === 'string' && /[gimuy]+/.test(key)) {
```
Comment (Pizzacus) [inline src/utils/reg.js]: Whoops that should be a `/^[gimuys]+$/`, sorry :x

File: src/commands/bday.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
+const sleep = require('util').promisify(setTimeout);
+const {Command} = require('discord-akairo');
+const reg = require('../utils/reg');
+const {sataniaName, extra} = require('../utils/regex-constants');
+
+const happyBirthday = /happy (birthday|bday|day of the birth)/;
```
Comment (Pizzacus) [inline src/commands/bday.js]: `\s*` and maybe make "the" optional

File: src/commands/praise.js (hunk 3 above)
Comment (Luffyyy) [reply src/commands/praise.js]: Hmm not initially I think but seems like that 's a good idea

**Pull request merged**